Kentor.AuthServices/IdentityProvider.cs

```diff
@@ -262,6 +263,19 @@ public AsymmetricAlgorithm SigningKey
 
         }
 
+        private X509Certificate2 certificate;
+
+        /// <summary>
+        /// The idp certificate stored locally, if any
+        /// </summary>
+        public X509Certificate2 Certificate
+        {
+            get
+            {
+                return certificate;
+            }
+        }
+

```

Comment:
I'm not happy with this part, but could not figure out another way to get the info needed by the TokenResolver. If there's a way to construct `SecurityToken` from just the `SigningKey`, let me know.